server/cluster_info.go

```diff
@@ -78,6 +79,38 @@ func loadClusterInfo(id core.IDAllocator, kv *core.KV, opt *scheduleOption) (*cl
 	return c, nil
 }
 
+func (c *clusterInfo) OnStoreVersionChange() {
+	var (
+		minVersion     *semver.Version
+		clusterVersion semver.Version
+	)
+
+	clusterVersion = c.opt.loadClusterVersion()
+	stores := c.GetStores()
+	for _, s := range stores {
+		if s.IsTombstone() {
+			continue
+		}
+		v := MustParseVersion(s.GetVersion())
+
+		if minVersion == nil || v.LessThan(*minVersion) {
+			minVersion = v
+		}
+	}
+	if clusterVersion.LessThan(*minVersion) {
```

Comment:
can we meet the case that PD itself's version is less than the min version?

Reply:
Yes. but the PD just provides currently knows features. so cluster still running with lower version feature. I think it's normal behavior because the cluster upgrades not finished. Maybe I need to warn it.

Reply:
em, if PD's version is older, it should not be able to provide service for the features that it does not have.